### ISSUE TITLE
active connection time via the websocket

### ISSUE THREAD

**Issue by koinju** [Bittrex/bittrex.github.io#230]
Hello, I am trying by all means to keep the connection active to wss://socket-v3.bittrex.com/signalr but impossible. It won't last more than 2 or 4 hours. We have a hypothesis that the bittrex server puts the socket in a blocked state which prevents the TCP client from reconnecting or cloudflare blocks our reconnection attempts. Any idea ?

**Comment by phillijw** [comment]
I've been having the same problem with orderbooks. Trades have been working fine though

**Comment by bittrex-pete** [comment]
@koinju I don't believe there's anything in place that would block connection attempts. Losing the connection once or twice a day wouldn't be surprising. Some customers of ours who are sensitive to drops will run multiple connections in parallel for redundancy. The other thing you can do which may help is move your code as close as possible to where we run our infrastructure which is in Azure West Europe. 

@phillijw What symptoms are you seeing? It's strange that you would have connection problems with one stream but not another.

**Comment by phillijw** [comment]
Over the weekend we simply stopped receiving messages on all orderbook channels. If I restarted the app, they would start coming in again. If I just let it sit there, they would also start coming in again, sometimes several hours later. It's possible this was caused by something wrong on my end but I can't find any evidence of that.

**Comment by jeffreytai** [comment]
> Over the weekend we simply stopped receiving messages on all orderbook channels. If I restarted the app, they would start coming in again. If I just let it sit there, they would also start coming in again, sometimes several hours later. It's possible this was caused by something wrong on my end but I can't find any evidence of that.

did you ever find out what the issue was? we experience this every several hours and have only been able to remedy it by setting a threshold for when the last orderbook message was received. the connection stays up and responds to TCP keep alives, i've run packet captures to show this as well

@bittrex-pete any ideas? we're by no means a random individual building a feed connection, we've built out 30+ different crypto exchange integrations for an institutional platform. bittrex is the only one with this consistent issue of just stopping feeds randomly

**Comment by Dimasik007** [comment]
hey, having the same issue with bittrex websockets where data just stops coming in. Also, integrated many exchanges before but for the first time have an issue like that. Any suggestions on what could be done?

**Comment by jeffreytai** [comment]
> hey, having the same issue with bittrex websockets where data just stops coming in. Also, integrated many exchanges before but for the first time have an issue like that. Any suggestions on what could be done?  

I opted for checking the current order book versus the one I had in memory when no updates have been received for some threshold, computed by the average arrival rate of price updates. If they didn't match up, I'd restart my connection

**Comment by Dimasik007** [comment]
thanks for your reply, I am just counting time since last message and use that to reconnect but feel like there could be a better way aha

**Comment by jeffreytai** [comment]
> thanks for your reply, I am just counting time since last message and use that to reconnect but feel like there could be a better way aha

yeah the annoying part is that the bittrex server still responds to TCP keep alives even when the feed randomly stops updating
